packages/amplify-category-analytics/src/commands/analytics/remove.ts

```diff
@@ -35,8 +17,19 @@ export const run = async (context: $TSContext): Promise<void> => {
   const { amplify, parameters } = context;
   const resourceName = parameters.first;
 
-  await amplify.removeResource(context, category, resourceName, { headless: false });
-  return removeResourceDependencies(context, resourceName);
-};
+  const blockRemovalOfInUseAnalytics = async (selectedAnalyticsResource: string): Promise<void> => {
+    const isResourceInUse = await checkResourceInUseByNotifications(context, selectedAnalyticsResource);
+    if (isResourceInUse) {
+      throw new AmplifyError('ResourceInUseError', {
+        message: `Analytics resource ${selectedAnalyticsResource} is being used by the notifications category and cannot be removed`,
+        resolution: `Run 'amplify remove notifications', then retry removing analytics`,
+      });
+    }
+  };
 
-export const name = subcommand;
+  // check if the CLI input analytics name is in use by notification
+  await blockRemovalOfInUseAnalytics(resourceName);
```

Comment:
nit: `throwIfUsedByAnalytics`

Reply:
This is the analytics category blocking removal of itself if it's used by notifications, not blocking some other resource if it's used by analytics

Reply:
`throwIfUsedByNotifications` then ? There's very similar function in auth remove. I'd follow the `throw` naming, i.e. `throw<why>`

Reply:
okay I'll rename